File: TicTacToe/app/src/main/java/hu/bme/aut/android/tictactoe/MainActivity.java
package hu.bme.aut.android.tictactoe;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import hu.bme.aut.android.tictactoe.model.TicTacToeModel;
import hu.bme.aut.android.tictactoe.ui.TicTacToeView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Button btnClear = findViewById(R.id.btnClear);
        final TicTacToeView ticTacToeView = findViewById(R.id.ticTacToe);

        btnClear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ticTacToeView.clearBoard();
            }
        });
    }


    public void showMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }


}

File: TicTacToeKotlin/app/src/main/java/hu/bme/aut/android/tictactoekotlin/MainActivity.kt
package hu.bme.aut.android.tictactoekotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.GridLayout
import hu.bme.aut.android.tictactoekotlin.data.Car
import hu.bme.aut.android.tictactoekotlin.data.Ship
import hu.bme.aut.android.tictactoekotlin.data.TicTacToeModel
import hu.bme.aut.android.tictactoekotlin.view.TicTacToeView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.concurrent.locks.Lock

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnRestart.setOnClickListener {
            ticTacToeView.resetGame()
        }
    }

    private fun TicTacToeView.resetGame() {
        TicTacToeModel.resetModel()
        invalidate()
    }

}
